Source/WebGPU/WGSL/WGSL.cpp

```diff
@@ -58,8 +80,25 @@ SuccessfulCheck::~SuccessfulCheck() = default;
 PrepareResult prepare(AST::ShaderModule& ast, const HashMap<String, PipelineLayout>& pipelineLayouts)
 {
     UNUSED_PARAM(pipelineLayouts);
-    Metal::RenderMetalCode metalCode = Metal::generateMetalCode(ast);
-    return { metalCode.metalSource.toString(), { } };
+    PhaseTimes phaseTimes;
+    Metal::RenderMetalCode generatedCode;
+
+    {
+        PhaseTimer phaseTimer("prepare total", phaseTimes);
+
+        RUN_PASS(resolveTypeReferences, ast);
+        RUN_PASS(rewriteEntryPoints, ast);
```

Comment:
Shouldn't this call be in a MSL-specific area? If we add a second backend code generator, surely that second backend wouldn't want this MSL-specific rewriting, right?

Reply:
I guess Metal-specific, not MSL, right? But in any case, we can always conditionally remove the pass later. I'm just trying to keep things as simple as possible (within reason) for now.